src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/OptimizedVSCompletionList.cs

```diff
@@ -32,6 +33,12 @@ protected override void WriteCompletionListProperties(JsonWriter writer, Complet
                     serializer.Serialize(writer, vsCompletionList.CommitCharacters);
                 }
 
+                if (vsCompletionList.Data != null)
+                {
+                    writer.WritePropertyName("_vsext_data");
+                    serializer.Serialize(writer, vsCompletionList.Data);
```

Comment:
Same question I had in a different PR, does this handle all the complexities of writing a property value properly?

Reply:
Yup, serialization deconstructs the object and turns it into json form (a string)